### PR TITLE
Added support for user search and user role grants.

### PR DESCRIPTION
**Pull request by jjh-ssh**

### CODE REVIEW

File: examples/example-role-grant.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import config
+import datetime
+
+# Initialize the API.
+api = privx_api.PrivXAPI(config.HOSTNAME, config.HOSTPORT, config.CA_CERT,
+                         config.OAUTH_CLIENT_ID, config.OAUTH_CLIENT_SECRET)
+
+# Authenticate.
+# NOTE: fill in your credentials from secure storage, this is just an example
+api.authenticate("7f577418-e543-4089-513d-71f9768b4589", "nGcNpvRHEeLaoFMO12FWiHb4JYdZdVRCv4lqCZBVL00=")
```
[the remark this answers is not in the document]
Comment (jjheinon) [reply examples/example-role-grant.py]: Fixed

File: privx_api/privx_api.py [hunk 2]
```diff
@@ -175,6 +176,19 @@ def _http_get(self, urlname: str) -> http.client.HTTPResponse:
 
         return conn.getresponse()
 
+    def _http_get_with_param(self, urlname: str, elem_id: str, data: dict = {}) -> http.client.HTTPResponse:
```
[the remark this answers is not in the document]
Comment (jjheinon) [reply privx_api/privx_api.py]: Fixed.

File: privx_api/privx_api.py [hunk 3]
```diff
@@ -32,6 +31,8 @@
 
     "rolestore.roles": "/role-store/api/v1/roles",
     "rolestore.sources": "/role-store/api/v1/sources",
+    "rolestore.users.search": "/role-store/api/v1/users/search",
+    "rolestore.user.roles": "/role-store/api/v1/users/{}/roles",
```
[the remark this answers is not in the document]
Comment (jjheinon) [reply privx_api/privx_api.py]: No time to tune this now, maybe later.

[the remark this answers is not in the document]
Comment (jjheinon) [reply privx_api/privx_api.py]: Added support for multiple custom params, removed http_get_with_param

File: privx_api/privx_api.py [hunk 4]
```diff
@@ -166,29 +166,18 @@ def _get_headers(self) -> dict:
...
-            headers=self._get_headers(),
-        )
-
-        return conn.getresponse()
-
-    def _http_get_with_param(self, urlname: str, elem_id: str) -> http.client.HTTPResponse:
-
+    def _http_get(self, urlname: str, params: dict = {}) -> http.client.HTTPResponse:
         conn = self._get_connection()
+        url = self._get_url(urlname).format(**params)
```
Comment (jukeks) [inline privx_api/privx_api.py]: This doesn't allow passing query parameters, only path parameters.

Comment (jjheinon) [reply privx_api/privx_api.py]: query params are in URL definition:
"rolestore.users.search": "/host-store/api/v1/hosts/search?offset={offset}&limit={limit}&sortkey={sortkey}&filter={filter}",

For path params:
 "rolestore.user.roles": "/role-store/api/v1/users/{user_id}/roles",

Comment (jukeks) [reply privx_api/privx_api.py]: That wouldn't work generally, as url parameters need to be url encoded. Also I don't think query parameters should be in that url definition.

**Comment by cuongssh** [comment]
inactive work, now incompatible with latest changes

**Pull request closed without merging**